common/src/main/java/tachyon/security/authentication/AuthenticationFactory.java

```diff
@@ -52,13 +54,23 @@
 
     private final String mAuthType;
 
-    AuthTypes(String authType) {
+    AuthType(String authType) {
       mAuthType = authType;
     }
 
     public String getAuthName() {
       return mAuthType;
     }
+
+    /**
+     * Validate the authentication type string and return a corresponding AuthType.
+     * @param authTypeStr authentication type string from configuration
+     * @return the corresponding AuthType
+     * @throws java.lang.IllegalArgumentException if the string does not match any type
+     */
+    public static AuthType getValidAuthType(String authTypeStr) {
+      return AuthType.valueOf(authTypeStr.toUpperCase(Locale.ENGLISH));
```

Comment:
The error message is `No enum constant tachyon.security.authentication.AuthenticationFactory.AuthType.WRONG`. I missed using `mThrown.expectMessage()` in the test.

I think it is good to add a more specific error message. Thanks. Currently it is not clear enough. Updated in new patch.